packages/build-utils/src/prerender.ts

```diff
@@ -117,10 +117,14 @@ export class Prerender {
         experimentalBypassFor.some(
           field =>
             typeof field !== 'object' ||
-            // host doesn't need a key
-            (field.type !== 'host' && typeof field.key !== 'string') ||
             typeof field.type !== 'string' ||
-            (field.value !== undefined && typeof field.value !== 'string')
+            (field.type === 'host' && 'key' in field) ||
+            (field.type !== 'host' && typeof field.key !== 'string') ||
+            (field.value !== undefined &&
+              typeof field.value !== 'string' &&
+              (typeof field.value !== 'object' ||
+                field.value === null ||
+                Array.isArray(field.value)))
```

Comment:
Might be nice to just break these into separate if conditions and validate the individual new conditions expected values but can be done separate.